rollout/trafficrouting/traefik/traefik.go

```diff
@@ -0,0 +1,164 @@
+package traefik
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/utils/defaults"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// Type holds this controller type
+const Type = "Traefik"
+
+var (
+	apiGroupToResource = map[string]string{
+		"traefik.containo.us": "traefikservices",
```

Comment:
You can use the `defaults.DefaultTraefikAPIGroup` for the key. It would be good to have the value (`traefikservices`) defined as a constant as well